kani-driver/src/assess.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use std::{collections::HashMap, path::PathBuf};
+
+use anyhow::Result;
+use comfy_table::Table;
+use kani_metadata::KaniMetadata;
+
+use crate::harness_runner::HarnessResult;
+use crate::session::KaniSession;
+
+/// Set some defaults for how we format tables
+fn assess_table_new() -> Table {
+    use comfy_table::*;
+
+    let mut table = Table::new();
+    table.set_content_arrangement(ContentArrangement::Dynamic);
+    table
+        .load_preset(comfy_table::presets::NOTHING)
+        .set_style(TableComponent::BottomBorder, '=')
+        .set_style(TableComponent::BottomBorderIntersections, '=')
+        .set_style(TableComponent::TopBorder, '=')
+        .set_style(TableComponent::TopBorderIntersections, '=')
+        .set_style(TableComponent::HeaderLines, '-')
+        .set_style(TableComponent::MiddleHeaderIntersections, '+')
+        .set_style(TableComponent::VerticalLines, '|');
+    table
+}
+
+fn unsupported_features_table(metadata: &KaniMetadata) -> Table {
+    // Map "unsupported feature name" -> (crates impacted, instance of use)
+    let mut counts: HashMap<String, (usize, usize)> = HashMap::new();
+
+    for item in &metadata.unsupported_features {
+        // key is unsupported feature name
+        let mut key = item.feature.clone();
+        // There are several "feature for <instance of use>" unsupported features.
+        // We aggregate those here by reducing it to just "feature"
+        if let Some((prefix, _)) = key.split_once(" for ") {
```

Comment:
Can we just use an enum in `unsupported_features` rather than relying on parsing strings which might change?